lib/src/map/options/enabled_gestures.dart

```diff
@@ -0,0 +1,287 @@
+      doubleTapZoomIn:
+          InteractiveFlag.hasFlag(flags, InteractiveFlag.doubleTapZoomIn),
+      doubleTapDragZoom:
+          InteractiveFlag.hasFlag(flags, InteractiveFlag.doubleTapDragZoom),
+      scrollWheelZoom:
+          InteractiveFlag.hasFlag(flags, InteractiveFlag.scrollWheelZoom),
+      twoFingerRotate:
+          InteractiveFlag.hasFlag(flags, InteractiveFlag.twoFingerRotate),
+      keyTriggerDragRotate:
+          InteractiveFlag.hasFlag(flags, InteractiveFlag.keyTriggerDragRotate),
+      trackpadZoom:
+          InteractiveFlag.hasFlag(flags, InteractiveFlag.trackpadZoom),
+    );
+  }
+
+  /// Enable panning with a single finger or cursor
+  final bool drag;
+
+  /// Enable fling animation after panning if velocity is great enough.
+  final bool flingAnimation;
+
+  /// Enable panning with multiple fingers
+  final bool twoFingerMove;
+
+  /// Enable zooming with a multi-finger pinch gesture
+  final bool twoFingerZoom;
+
+  /// Enable rotation with two-finger twist gesture
+  final bool twoFingerRotate;
+
+  /// Enable zooming with a single-finger double tap gesture
+  final bool doubleTapZoomIn;
+
+  /// Enable zooming with a single-finger double-tap-drag gesture
+  ///
+  /// The associated [MapEventSource] is [MapEventSource.doubleTapHold].
+  final bool doubleTapDragZoom;
+
+  /// Enable zooming with a mouse scroll wheel
+  final bool scrollWheelZoom;
+
+  /// Enable zooming with the device trackpad / touchpad
+  final bool trackpadZoom;
+
+  /// Enable rotation by pressing the defined keyboard key (by default CTRL key)
+  /// and dragging with the cursor
+  /// or finger.
+  final bool keyTriggerDragRotate;
+
+  /// Returns true of any gesture with more than one finger is enabled.
+  bool hasMultiFinger() => twoFingerMove || twoFingerZoom || twoFingerRotate;
+
+  /// Wither to change the value of some gestures. Returns a new
+  /// [EnabledGestures] object.
+  EnabledGestures copyWith({
+    bool? drag,
+    bool? flingAnimation,
+    bool? twoFingerZoom,
+    bool? twoFingerMove,
+    bool? doubleTapZoomIn,
+    bool? doubleTapDragZoom,
+    bool? scrollWheelZoom,
+    bool? twoFingerRotate,
+    bool? keyTriggerDragRotate,
+    bool? trackpadZoom,
+  }) =>
+      EnabledGestures(
+        drag: drag ?? this.drag,
+        flingAnimation: flingAnimation ?? this.flingAnimation,
+        twoFingerZoom: twoFingerZoom ?? this.twoFingerZoom,
+        twoFingerMove: twoFingerMove ?? this.twoFingerMove,
+        doubleTapZoomIn: doubleTapZoomIn ?? this.doubleTapZoomIn,
+        doubleTapDragZoom: doubleTapDragZoom ?? this.doubleTapDragZoom,
+        scrollWheelZoom: scrollWheelZoom ?? this.scrollWheelZoom,
+        twoFingerRotate: twoFingerRotate ?? this.twoFingerRotate,
+        keyTriggerDragRotate: keyTriggerDragRotate ?? this.keyTriggerDragRotate,
+        trackpadZoom: trackpadZoom ?? this.trackpadZoom,
+      );
+
+  const EnabledGestures.byGroup({
```

Comment:
Maybe this should have optional params, `true` by default? Would allow removal of `noRotation` constructor. WDYT?

Reply:
the name `byGroup` would be a disable-all-by-default constructor for my understanding. It wouldn't be reasonable for me that calling this constructor without parameters would enable all gestures.
Default parameter values are a great thing but in this case its 3 boolean parameters we can assume the user wants to set at least one anyways when using it.

Reply:
> we can assume the user wants to set at least one anyways when using it

Yeah, so if they want to enable one, then they need to set the others as well. If they want to disable one, they need to set the others as well. If they want to enable two, they still need to disable the last, and vice versa. I think this just backs up my argument more.

> the name byGroup would be a disable-all-by-default constructor for my understanding. It wouldn't be reasonable for me that calling this constructor without parameters would enable all gestures.

OK, then we can default to `false`. Or rename to `disableByGroup`. Either way, I feel pretty strongly about this.

Reply:
Optimally we double have both to let users decide if they want to follow a whitelist or blacklist approach. We could do a similar design as the individual gestures by providing a `byGroup`, `disableByGroup` and `enableByGroup` constructor. 

Otherwise I'd prefer false as default value for the `byGroup` constructor.

What do you prefer?

Reply:
Ok, let's support all 3. Although we should keep it consistent between `all`/`none` and `enabled`/`disabled`: it's up to you which you think is better.

Reply:
Then remove `noRotation`?

Reply:
> Ok, let's support all 3. Although we should keep it consistent between all/none and enabled/disabled: it's up to you which you think is better.

Added `allByGroup` and `noneByGroup` contructors.

> Then remove noRotation?

Still no fan of this shortcut constructor, huh? 😄 Alright, I removed it.